### PR TITLE
Implement Inspect command

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
I'm adding a new command, `tuist inspect` to print information for any graph node.

### Implementation 👩‍💻👨‍💻
- [x] Implement the command
- [ ] Add tests
- [ ] Add documentation
- [ ] Update the CHANGELOG

image

### CODE REVIEW

File: Sources/TuistKit/Services/InspectService.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        logger.info(" - Product: \(target.target.product.rawValue)")
+        logger.info(" - Bundle identifier: \(target.target.bundleId)")
+        logger.info(" - Static: \(target.target.product.isStatic)")
+        logger.info(" - Sources: \(target.target.sources.count)")
+        logger.info(" - Resources: \(target.target.resources.count)")
+
+        logger.info("Dependencies:", metadata: .subsection)
+        logger.info(" - Dependencies: \(allDependencies.count)")
+        logger.info(" - Direct dependencies: \(target.dependencies.count)")
+        logger.info(" - Transitive dependencies: \(allDependencies.count - target.dependencies.count)")
```
Comment (andreacipriani) [inline Sources/TuistKit/Services/InspectService.swift:79]: Does this work only for Swift and linked frameworks?
 
If `A` depends on `B` via a header exposed by search paths and `B` depends on `C`,`D`,`E` via other headers, I don't think that `Transitive dependencies` of `A` would return `C`, `D` and `E` as it should

File: Sources/TuistKit/Commands/InspectCommand.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+struct InspectCommand: ParsableCommand {
+    static var configuration: CommandConfiguration {
+        CommandConfiguration(commandName: "inspect",
+                             abstract: "Inspects a target from the dependency graph and prints information about it.")
+    }
+
+    // MARK: - Options
+
+    @OptionGroup()
+    var options: DocCommand.Options
```
Comment (laxmorek) [inline Sources/TuistKit/Commands/InspectCommand.swift:17]: `var options: DocCommand.Options` -> `var options: InspectCommand.Options`

Please correct me if Im missing something. 🤔

File: Sources/TuistKit/Commands/InspectCommand.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    // MARK: - Run
+
+    func run() throws {
+        let absolutePath: AbsolutePath
+        if let path = options.path {
+            absolutePath = AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+        } else {
+            absolutePath = FileHandler.shared.currentPath
+        }
```
Comment (laxmorek) [inline Sources/TuistKit/Commands/InspectCommand.swift:27]: It is not related to this PR but I would like to mention it:
1. Defining path is duplicated across commands implementations.
2. Some commands like: `Dependencies`, `Up`, `Generate` contain this logic into theirs services (`*Service`). Some others like `Doc`, `Test` and this one have this logic here (`*Command`).

Comment (laxmorek) [reply Sources/TuistKit/Commands/InspectCommand.swift:27]: ad.1 Can we somehow abstract this logic? 
ad.2 Should we be consistent or this case is not important.

**Pull request closed without merging**